### PR TITLE
GrafanaUI: Add disabled option for menu items

### PR DESCRIPTION
**Pull request by going-confetti**
**What is this feature?**
This feature allows developers to disable individual menu items and submenus of the `Menu` component.

**Why do we need this feature?**

See https://github.com/grafana/grafana/discussions/58853 for more details

**Who is this feature for?**
Developers building apps that use `@grafana/ui`

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -200,6 +221,17 @@ const getStyles = (theme: GrafanaTheme2) => {
         }
       }
     `,
+    disabled: css`
+      color: ${theme.colors.action.disabledText};
```
Comment (going-confetti) [inline packages/grafana-ui/src/components/Menu/MenuItem.tsx:225]: I could use some help here:
Will the color alone be enough or should the background be changed as well?
image
image

There's also `theme.colors.action.disabledOpacity` which results into this:
image

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Nice job 👏 looks great, just a couple of small things on the story

File: packages/grafana-ui/src/components/Menu/Menu.story.tsx [hunk 2]
```diff
@@ -79,6 +104,7 @@ export function Examples() {
           <Menu.Item
             label="item2"
             icon="apps"
+            disabled
```
[the remark this answers is not in the document]
Comment (going-confetti) [reply packages/grafana-ui/src/components/Menu/Menu.story.tsx]: Oops 👀, I'm glad you noticed! Should be fixed now

File: packages/grafana-ui/src/components/Menu/Menu.story.tsx [hunk 3]
```diff
@@ -49,6 +49,31 @@ export function Examples() {
           <Menu.Item label="With destructive prop set" icon="trash-alt" destructive />
         </Menu>
       </StoryExample>
+      <StoryExample name="With disabled items">
+        <Menu>
+          <Menu.Item label="Google" icon="search-plus" />
+          <Menu.Item label="Disabled action" icon="history" disabled />
+          <Menu.Item label="Disabled link" icon="external-link-alt" disabled />
```
[the remark this answers is not in the document]
Comment (going-confetti) [reply packages/grafana-ui/src/components/Menu/Menu.story.tsx]: Thank you, totally forgot about it

**Review by JoaoSilvaGrafana — APPROVED** [review]
Looks good to me! 🎉

**Review by mckn — APPROVED** [review]
LGTM! Great work on this.

**Pull request merged**